### ISSUE TITLE
Some issue about training loss

### ISSUE THREAD

**Issue by bianzheng123** [twistedcubic/learn-to-hash#5]
Hi
Thanks for code sharing. I have reproduced the code in SIFT1M. I used neural network for prediction and I found the loss only decline a little bit, from 69 to 57 in 20 epochs.
Coule you please explain about this phenomenon?
![Snipaste_2020-12-14_16-01-55](https://user-images.githubusercontent.com/51192378/102055374-c123e800-3e25-11eb-9676-be2087289100.png)

**Comment by twistedcubic** [comment]
Hi @bianzheng123,

Glad you are running the code and are working with the SIFT dataset.

There are several factors that affect the training loss. The key metric we used is to ensure that *validation* accuracy increases steadily.

I don't recall the exact training loss values we had for each setting (and they are different depending on the number of parts and levels), but we deliberately added regularizers, dropouts, and learning rate schedulers to mitigate overfitting, i.e. not allowing the training loss to drop precipitously. When the training loss got too low, it usually indicated overfitting.
